### ISSUE TITLE
File Operations "Handle" Parameter Int32 Issue

### ISSUE THREAD

**Issue by datruongMS** [gbrueckl/Databricks.API.PowerShell#37]
We are using 1.6.0.0 and tried 1.7.0.0 versions of DatabricksPS and are running into parameter size issues. No change on our side, no version upgrade, etc. and it started appearing 4/20/21.

Specifically, we are using "Upload-DatabricksFSFile" and providing parameters "-Path $... -LocalPath $... -Overwrite $true"

```
Upload-DatabricksFSFile : Cannot process argument transformation on parameter 'Handle'. Cannot convert value 
"2265311117716249771" to type "System.Int32". Error: "Value was either too large or too small for an Int32."
At ...
+ Upload-DatabricksFSFile -Path $... -LocalPath $... ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Upload-DatabricksFSFile], ParameterBindingArgumentTransformationExcept 
   ion
    + FullyQualifiedErrorId : ParameterArgumentTransformationError,Upload-DatabricksFSFile
```

There is no "Handle" parameter on the command "Upload-DatabricksFSFile" but it internally uses other commands like "Add-DatabricksFSFile" which does specifiy a "Handle" parameter. I believe the file handle is too large to fit into an Int32 and is causing us to not be able to run the command.

Can some guidance be provided on this issue?

**Comment by datruongMS** [comment]
I can confirm that the handle being returned is too large to fit into an Int32. After turning on the "-Verbose" parameter on "Upload-DatabricksFSFile", we get this response:

```2021-04-21T17:54:13.9764217Z VERBOSE: Creating new file in DBFS at dbfs:...
2021-04-21T17:54:13.9811682Z VERBOSE: Building Body/Parameters for final API call ...
2021-04-21T17:54:13.9834685Z VERBOSE: Checking if Databricks environment has been initialized yet ...
2021-04-21T17:54:13.9846720Z VERBOSE: Databricks environment already initialized!
2021-04-21T17:54:13.9858883Z VERBOSE: Setting final ApiURL ...
2021-04-21T17:54:13.9875260Z VERBOSE: Getting Headers for Databricks API call ...
2021-04-21T17:54:13.9897960Z VERBOSE: API Call: POST ...
2021-04-21T17:54:13.9909901Z VERBOSE: Building Headers ...
2021-04-21T17:54:13.9925528Z VERBOSE: Getting Headers for Databricks API call ...
2021-04-21T17:54:13.9944486Z VERBOSE: Found a null-Value to add as Content-Type ...
2021-04-21T17:54:13.9970987Z VERBOSE: null-value is omitted.
2021-04-21T17:54:13.9990362Z VERBOSE: Found a null-Value to add as Accept ...
2021-04-21T17:54:14.0003641Z VERBOSE: null-value is omitted.
2021-04-21T17:54:14.0048696Z VERBOSE: Headers: 
2021-04-21T17:54:14.0064725Z 
2021-04-21T17:54:14.0072817Z 
2021-04-21T17:54:14.0087221Z Name                           Value                                                                                   
2021-04-21T17:54:14.0094263Z 
2021-04-21T17:54:14.0101579Z 
2021-04-21T17:54:14.0113241Z ----                           -----                                                                                   
2021-04-21T17:54:14.0120545Z 
2021-04-21T17:54:14.0128442Z 
2021-04-21T17:54:14.0140790Z Authorization                  Bearer ***                                             
2021-04-21T17:54:14.0146501Z 
2021-04-21T17:54:14.0159940Z 
2021-04-21T17:54:14.0171903Z Content-Type                   application/json                                                                        
2021-04-21T17:54:14.0179174Z 
2021-04-21T17:54:14.0185822Z 
2021-04-21T17:54:14.0192770Z 
2021-04-21T17:54:14.0199840Z 
2021-04-21T17:54:14.0206937Z 
2021-04-21T17:54:14.0218471Z VERBOSE: POST request - converting Body to JSON ...
2021-04-21T17:54:14.0229822Z VERBOSE: Body: 
2021-04-21T17:54:14.0241305Z {
2021-04-21T17:54:14.0248333Z 
2021-04-21T17:54:14.0261219Z     "path":  "...",
2021-04-21T17:54:14.0267700Z 
2021-04-21T17:54:14.0279124Z     "overwrite":  true
2021-04-21T17:54:14.0285758Z 
2021-04-21T17:54:14.0297163Z }
2021-04-21T17:54:14.0308437Z VERBOSE: POST ... with -1-byte payload
2021-04-21T17:54:14.7010981Z VERBOSE: received -1-byte response of content type application/json
2021-04-21T17:54:14.7026924Z VERBOSE: Response: {
2021-04-21T17:54:14.7034409Z 
2021-04-21T17:54:14.7048644Z     "handle":  **2265311096011116773**
2021-04-21T17:54:14.7056256Z 
2021-04-21T17:54:14.7069932Z }
```

**Comment by gbrueckl** [comment]
Ok, i know what the issue is and will fix it on Friday
Thanks for reporting!

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by gbrueckl** [comment]
@datruongMS I just fixed the issue and publish the new version 1.8.0.1 to the gallery

kindly let me know if it works for you now 

and thanks again for reporting the issue!

**Comment by gbrueckl** [comment]
closing as it would fixed and not reported again
